experimental/Schemes/SpaceGerms.jl

```diff
@@ -48,13 +58,141 @@ A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X``
   end
 end
 
+@doc raw"""
+    HypersurfaceGerm{BaseRingType, RingType, SpecType}
+A hypersurface germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct HypersurfaceGerm{
+                 BaseRingType<:Ring,
+                 RingType<:Ring,
+                 SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+  f::RingElem
+
+  function HypersurfaceGerm(X::GermAtClosedPoint,f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    if check
```

Comment:
```suggestion
    @check begin
```
We should use this macro in general.

Reply:
done